### ISSUE TITLE
Referencing another column

### ISSUE THREAD

**Issue by ktwbc** [vitalets/x-editable-yii#10]
Is there a way to have one column use another column's value? For example, I have two columns, one for country and one for state/province. I need the second column's source to reference the value of the first so I can return only the states in that country.

It appears that the $data keyword you would normally use isn't available and I've tried everything including using a js function for the source instead of a CreateUrl but no luck.

```
  'columns'=>array(
        array(
            'class' => 'editable.EditableColumn',
            'name' => 'country',
            'headerHtmlOptions' => array('style' => 'width: 110px'),
            'editable' => array(
                'type' => 'select',
                'url' => $this->createUrl('shipping/updateDestination'),
                'placement' => 'right',
                'source' => Country::getCountriesForTaxes(),
                'emptytext'=> 'disabled',
                'options'=>array(
                    'onblur' => 'submit',
                    'success' => 'js:function(response, newValue) {
                                if(response=="delete")
                                    $.fn.yiiGridView.update("user-grid");
                            }',
                    'showbuttons' => false,
                    'validate' => 'js: function(value) {
                    if($.trim(value) == "") return "This field is required";
                    }'
                ),
            )
        ),
        array(
            'class' => 'editable.EditableColumn',
            'name' => 'state',
            'headerHtmlOptions' => array('style' => 'width: 110px'),
            'editable' => array(
                'url' => $this->createUrl('shipping/updateDestination'),
                'type' => 'select',
                'source'=>CHtml::listData(State::model()->findAll(), 'id', 'state'),
                'placement' => 'right',
                'options'=>array(
                    'onblur' => 'submit',
                    'showbuttons' => false,
                ),
            )
        ),
```

**Comment by ktwbc** [comment]
I'm actually trying to get at the $data keyword which you can normally access in a CViewGrid but it doesn't appear that's accessible in x-editable? Something like

```
 'source'=>State::getStateList($data->countryid),
```

**Comment by vitalets** [comment]
it looks that source of `state` column better to be an url, not array -  as user can change the country and we need to update source.
Unfortunatly in current version `source` defined as function and returning string not treated as url. I will definetly fix in next release.
Currently the solution can be a bit tricky (see comments in code):

```
 'columns'=>array(
        array(
            'class' => 'editable.EditableColumn',
            'name' => 'country',
            'headerHtmlOptions' => array('style' => 'width: 110px'),
            'editable' => array(
                'type' => 'select',
                'url' => $this->createUrl('shipping/updateDestination'),
                'placement' => 'right',
                'source' => Country::getCountriesForTaxes(),
                'emptytext'=> 'disabled',
                'options'=>array(
                    'onblur' => 'submit',
                    'success' => 'js:function(response, newValue) {
                       var $state = $(this).closest('td').next().find('.editable'); // <-- get state element
                            $state.editable('option', 'source', '/getStates.php?country='+newValue); // <-- update source as country changed
                                if(response=="delete")
                                    $.fn.yiiGridView.update("user-grid");
                            }',
                    'showbuttons' => false,
                    'validate' => 'js: function(value) {
                    if($.trim(value) == "") return "This field is required";
                    }'
                ),
               'onInit' => 'js: function(e, params) {    // <--- we need to set source initially
                       var $state = $(this).closest('td').next().find('.editable'); 
                       var country = $(this).data().value;
                       $state.editable('option', 'source', '/getStates.php?country='+country); 
               }' 
            )
        ),
        array(
            'class' => 'editable.EditableColumn',
            'name' => 'state',
            'headerHtmlOptions' => array('style' => 'width: 110px'),
            'editable' => array(
                'url' => $this->createUrl('shipping/updateDestination'),
                'type' => 'select',
              //  'source'=>CHtml::listData(State::model()->findAll(), 'id', 'state'), //<-- no source here
                'placement' => 'right',
                'options'=>array(
                    'onblur' => 'submit',
                    'showbuttons' => false,
                ),
            )
        ),
```

**Comment by vitalets** [comment]
it should work in just released 1.1.0
